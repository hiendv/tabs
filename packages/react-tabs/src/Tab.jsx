import React from 'react'
import PropTypes from 'prop-types'

function Tab (props) {
  return (null)
}

Tab.propTypes = {
  title: PropTypes.string,
  icon: PropTypes.object,
  hash: PropTypes.string,
  end: PropTypes.bool,
  ghost: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.node
}

Tab.defaultProps = {
  title: 'Title',
  icon: {
    attrs () {},
    path () {}
  },
  hash: null,
  end: false,
  ghost: false,
  onClick () { },
  children: undefined
}

export default Tab
