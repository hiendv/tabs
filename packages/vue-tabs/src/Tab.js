export default {
  name: 'Tab',
  functional: true,
  props: {
    title: {
      type: String,
      default () {
        return 'Title'
      }
    },
    icon: {
      type: Object,
      default () {
        return {
          attrs () {},
          path () {}
        }
      }
    }
  },
  render (h, context) {
    return h(
      'div',
      context,
      context.scopedSlots.default({})
    )
  }
}
