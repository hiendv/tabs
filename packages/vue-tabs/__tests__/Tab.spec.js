import { shallowMount } from '@vue/test-utils'
import Tab from '../src/Tab.js'

describe('Tab', () => {
  it('renders', () => {
    const wrapper = shallowMount(Tab, {
      scopedSlots: {
        default () {
          return this.$createElement('div', {}, 'Gotcha')
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
