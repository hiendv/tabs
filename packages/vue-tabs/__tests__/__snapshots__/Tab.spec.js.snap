// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders 1`] = `
<div>
  <div>Gotcha</div>
</div>
`;
